RAGchain/preprocess/text_splitter/token_splitter.py

```diff
@@ -0,0 +1,103 @@
+import copy
+
+from typing import List
+from uuid import uuid4
+
+from langchain.schema import Document
+from langchain.text_splitter import CharacterTextSplitter, TokenTextSplitter, SpacyTextSplitter, SentenceTransformersTokenTextSplitter, NLTKTextSplitter
+from transformers import GPT2TokenizerFast
+from RAGchain.preprocess.text_splitter.base import BaseTextSplitter
+from RAGchain.schema import Passage
+
+
+class Token_Splitter(BaseTextSplitter):
+    """
+    Split a document into passages by recursively splitting on a list of separators.
+    You can specify a window_size and overlap_size to split the document into overlapping passages.
+    """
+    def __init__(self, tokenizer_name: str = 'tiktoken', chunk_size: int = 0, chunk_overlap: int = 0, **kwargs):
+        """
+        :param tokenizer_name: A tokenizer_name name. You can choose tokenizer_name.
+                        (tiktoken, spaCy, SentenceTransformers, NLTK, huggingFace)
+        :param chunk_size: Maximum size of chunks to return. Default is 0.
+        :param chunk_overlap: Overlap in characters between chunks. Default is 0.
+        :param kwargs: Additional arguments.
+        All splitters were inherited TextSplitter class in langchain text_splitter.py.
+        """
+        self.chosen_tokenizer = tokenizer_name
+
+        # tiktoken
+        self.tiktoken_splitter = TokenTextSplitter.from_tiktoken_encoder(chunk_size = chunk_size, chunk_overlap =  chunk_overlap)
+
+        # spaCy
+        self.spaCy_splitter = SpacyTextSplitter(chunk_size= chunk_size, chunk_overlap= chunk_overlap)
+
+        # SentenceTransformers (Default: chunk_overlap=0)
+        self.SentenceTransformers_splitter = SentenceTransformersTokenTextSplitter(chunk_size= chunk_size, chunk_overlap= chunk_overlap)
+
+        # NLTK
+        self.NLTK_splitter = NLTKTextSplitter(chunk_size= chunk_size, chunk_overlap= chunk_overlap)
+
+        # Hugging Face
+        tokenizer = GPT2TokenizerFast.from_pretrained("gpt2")
+        self.huggingFace_splitter = CharacterTextSplitter.from_huggingface_tokenizer(
+            tokenizer, chunk_size= chunk_size, chunk_overlap= chunk_overlap
+        )
```

Comment:
Please consider when the user wants to use custom huggingface tokenizer for splitter. 
It is really common use case for HuggingfaceTokenSplitter. 
Your code looks only using gpt2 tokenizer.